src/Services/QueuedJobService.php

```diff
@@ -892,18 +943,16 @@ public function runJob($jobId)
                     /** @var AbstractQueuedJob|QueuedJob $job */
                     $job->afterComplete();
                     $jobDescriptor->cleanupJob();
+
+                    $this->extend('updateJobDescriptorAndJobOnCompletion', $jobDescriptor, $job);
                 }
             } catch (Exception $e) {
-                // okay, we'll just catch this exception for now
-                $this->getLogger()->error(
-                    $e->getMessage(),
-                    [
-                        'exception' => $e,
-                    ]
-                );
-                $jobDescriptor->JobStatus =  QueuedJob::STATUS_BROKEN;
-                $this->extend('updateJobDescriptorAndJobOnException', $jobDescriptor, $job, $e);
-                $jobDescriptor->write();
+                // PHP 5.6 exception handling
+                $this->handleBrokenJobException($jobDescriptor, $job, $e);
+                $broken = true;
+            } catch (\Throwable $e) {
+                // PHP 7 exception handling
```

Comment:
Technically the majority of PHP 7 exceptions would be caught in the previous catch, right? Would it be more honest to say "PHP 7 Error handling" ?